### ISSUE TITLE
Please review CG Report requirements

### ISSUE THREAD

**Issue by ianbjacobs** [solid/specification#587]
Hi all,

In looking at a sample Solid technical report such as:
 https://solidproject.org/TR/protocol

I would like to draw the CG's attention to our CG Report requirements:
  https://www.w3.org/community/reports/reqs/

Among them notably, we have style requirements (to help people understand the status of documents) and copyright statement requirements (per the CG policies), etc.

I'd like to request that the CG make appropriate changes based on these requirements. Don't hesitate to contact me on team-community-process if you have questions.

Cheers,
Ian

[1] https://www.w3.org/community/about/process/summary/

**Comment by ianbjacobs** [comment]
Hi @sarven,

> CGs are not prohibited from producing documents that are not CG-DRAFT/FINAL, with any status, license, at any 
> location, or using any particular stylesheets/scripts (=we're aware that they're not or equivalent to W3C "Community 
> Submissions").

That is also my understanding.

> Deliverables of a WG charter or proposed specifications are not required to have a CG-DRAFT/FINAL status 
> in order for them to be accepted in Recommendation/Note/Registry Tracks.

That is also my understanding (but not directly relevant to the current thread about CG Report requirements).

> On a related note, the common understanding is that a "Working Group will not adopt new proposals until they 
> have matured through the [CG] or another similar incubation phase.". 

Here is what the Advisory Board wrote in 2016 [1] on that topic: "Strongly Recommended: Charters do not list specs as deliverables, and WGs do not publish FWPDs, until there is rough consensus across stakeholders that the spec solves a real problem, is likely to be implemented, and is likely to be used on the Web. This consensus could emerge from an incubation phase in WICG or another CG, or in a WG that has an established culture of taking and vetting suggestions from its public mailing list."

[1] https://www.w3.org/Guide/standards-track/#criteria

> We have taken steps to make sure that copyright/licensing in the CG charter will remain compatible with 
> W3C policies and have acknowledged that agreements made through CG-DRAFT/FINALs will better 
> help to transition works from the CG to a WG.

Our FAQ [2] includes this:

Q. Can I make it a condition of joining a Community or Business Group or publishing a Report that people must agree to licensing terms beyond the CLA/FSA?
A. No.

Is the expectation that people cannot join the Solid CG unless they agree to the terms and conditions of the charter?

[2] https://www.w3.org/community/about/faq/#can-i-make-it-a-condition-of-participation-in-a-community-or-business-group-that-people-must-agree-to-licensing-terms-beyond-the-clafsa

> Lastly, I understand the differences between what can be practised, required, or prohibited by W3C in a CG (and elsewhere) may be confusing to some or even many. I can only recommend that existing CG requirements, W3C Process, and other guides are updated to further help minimise these confusions for anyone, and to better communicate expectations. If you'd like me/us to follow-up on this with you / W3C Team in another thread/repository, happy to do so.

We welcome suggestions on team-community-process@w3.org. Thank you!

**Comment by michielbdejong** [comment]
@ianbjacobs the list of draft reports I filled in on https://www.w3.org/community/solid/ during our meeting today was a copy of https://solidproject.org/TR/#work-items

But since then I got a message from @VirginiaBalseiro saying this list is incorrect and not conforming to https://www.w3.org/community/reports/reqs/. I was also unaware that this had been an open issue since October.

As we discussed there is an 'add' button but no 'remove' button. Can you please remove the list again? Sorry for the hassle!

**Comment by csarven** [comment]
I will PR a CG-DRAFT (and eventually FINAL) of the Solid Protocol based on [ED](https://solidproject.org/ED/protocol). I have the CG-DRAFT ready but was waiting on the possible resolution of some of the items in the last milestone if the Group wanted them in there. There is no showstopper to have a CG-DRAFT of what we have, and we can go ahead with it. I can PR next in the week or so (keeping in mind travels). We (CG) can perhaps pick it up on 2024-05-08.

We can publish other Work Items as CG-DRAFT/FINALs once they're ready and agreed upon.

**Comment by ianbjacobs** [comment]
Hi all,

If these are indeed reports of the CG, then it is important that they adhere to the CG report requirements [1], including the copyright statement, use of logos, boilerplate text, etc. Let me know how I can help on this.

Regarding the link from the CG’s home page, I think we can leave them in place since the more important bit is that the spec styles be updated. (And I’m assuming the URLs will point at the updated drafts.)

Thank you,

Ian


[1] https://www.w3.org/community/reports/reqs/
[2] https://www.w3.org/community/solid/

> On Apr 25, 2024, at 9:57 AM, Sarven Capadisli ***@***.***> wrote:
> 
> 
> I will PR a CG-DRAFT (and eventually FINAL) of the Solid Protocol based on ED. I have the CG-DRAFT ready but was waiting on the possible resolution of some of the items in the last milestone if the Group wanted them in there. There is no showstopper to have a CG-DRAFT of what we have, and we can go ahead with it. I can PR next in the week or so (keeping in mind travels). We (CG) can perhaps pick it up on 2024-05-08.
> We can publish other Work Items as CG-DRAFT/FINALs once they're ready and agreed upon.
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

--
Ian Jacobs ***@***.***>
https://www.w3.org/People/Jacobs/
Tel: +1 917 450 8783

**Comment by VirginiaBalseiro** [comment]
> @ianbjacobs the list of draft reports I filled in on https://www.w3.org/community/solid/ during our meeting today was a copy of https://solidproject.org/TR/#work-items
> 
 @michielbdejong can you clarify what "our meeting" is referring to?

**Comment by csarven** [comment]
@ianbjacobs I think we already covered this earlier in the issue and elsewhere.

Just for clarity for those not heavily involved in the CG or who need a summary:

Our process prior to the [Solid CG Charter](https://www.w3.org/community/solid/charter/) didn't include publishing [W3C CG Reports](https://www.w3.org/community/reports/). The documents that are currently referenced from Solid [Work Items](https://solidproject.org/TR/#work-items) are not "W3C CG Reports" as it stands but we're transitioning to it. We have documents with status along the lines of editors/unofficial drafts, living documents, notes, and lowercase technical reports - was never intended to be conflated with [W3C Technical Reports](https://www.w3.org/2023/Process-20231103/#Reports), not hosted w3.org, not using W3C logo, not implying W3C endorsed, among other things to say the least.

---

@ianbjacobs :

* To (better) ensure persistence and immutability, can the CG publish CG-FINALs under w3.org? Would you recommend that we do?
* Are CG-DRAFTs intended to be mutable or immutable snapshots? As I understand it, there can be multiple CG-DRAFTs with different versions of a report.
* As mentioned, majority of the documents listed under our Work Items are drafts of sorts. So, related to the above question, should we consider all of the documents as potentially equivalent to CG-DRAFTs or should we only publicise what may be deemed to be mature / advanced enough to be CG-DRAFT (and later CG-FINALs)? For instance, in loose terms, the documents that are published under `https://solidproject.org/TR/` (e.g., https://solidproject.org/TR/protocol ) would qualify for CG-DRAFT, but I'm not sure how to approach the others in a fair way that both reflects their maturity as well as group's consensus.

Related:
* https://github.com/w3c/tr-pages/issues/102 - as discussed in the issue, this is rather important to distinguish ongoing WIP / drafts (along the lines of ED/UD) from CG-DRAFT publications.
* https://github.com/solid/specification/issues/373
* ...

**Comment by ianbjacobs** [comment]
@VirginiaBalseiro, @michielbdejong and I were chatting about CGs in the context of another group.

**Comment by ianbjacobs** [comment]
@csarven, @VirginiaBalseiro, and @michielbdejong,

If [these documents](https://solidproject.org/TR/#work-items) are not (yet) CG Reports, please do not indicate that they were published by the Solid Community Group. Instead, please indicate which individuals drafted them. Because these documents are not CG deliverables, please remove mention of the CLA. (Indeed, you refer to the MIT license in the documents that I looked at.)

In other words: these are either CG Reports published by the group under the CLA and consistent with the requirements for CG Reports, or they are some other kind of document that has no relationship to any W3C process or group.

Regarding your questions above:

* All Final Reports must be published on w3.org (per the CG process). The recommended way to do that is via the [cg-reports repo](https://github.com/w3c/cg-reports).
* CG drafts can be living documents; they do not need to be immutable (but can be).

**Comment by elf-pavlik** [comment]
> CG drafts can be living documents; they do not need to be immutable (but can be).

Most Solid drafts had versioned snapshots published as "WD" and living documents available as "ED".
Many documents use Bikeshed, [which seems only to have two options](https://speced.github.io/bikeshed/#metadata)—*CG-DRAFT* and *CG-FINAL*. I understand that no templates equivalent to ED and WD are available for Community Groups. Until finalized, should they use the *CG-DRAFT* template? We should still be able to have immutable releases similar to "WD" tagged with 0.x version numbers.

Linking from the 0.x tagged immutable version (~WD) to the latest version on the main branch (~ED) might be slightly nuanced since the link would commonly be labeled Editor's Draft. Is there a recommended term that could be used in CG drafts to label that link?

**Comment by ianbjacobs** [comment]
I am still getting up to speed on the status and plans for these documents. But if a document is not yet a Draft (or Final) CG Report, it should not claim to be the product of the CG (and not use the CG styles, license, etc.).  If a call would be useful, let's find time next week.

**Comment by ianbjacobs** [comment]
Yes I can join that call (please email me the call info). Thanks!

**Comment by csarven** [comment]
Quick status update:

I have the CG-DRAFT ready but decided not to PR yet. Traveling today. I will do it this week in a calmer moment/space. More details will be in the PR that we can review. A quick FYI for now:

* CG-DRAFT conforms to [CG Report Requirements](https://www.w3.org/community/reports/reqs/) (but of course to be confirmed by W3C Team Community Process folks).
* It also conforms to / be compatible with some of the [W3C publication rules](https://www.w3.org/pubrules/doc) (without interfering with CG Report Requirements), so it should be even workable (First Public) Working Draft or later states as needed. I've already went through this publication process with specs in WGs.
* CG-DRAFT can be published at https://solidproject.org/TR/protocol , with a link to "This version" for the specific snapshot (i.e., Version 0.11.0. e.g., `https://solidproject.org/TR/2024/{YYYYMMDD}-protocol`).
* Once we get passed that publication, we can look into releasing CG-FINAL. I'd still like to encourage the Group to communicate with implementation experience, e.g., who is implementing or does not want to implement N3 Patch, ditto IRIs,..., which can be used to make another CG-DRAFT and/or towards the CG-FINAL.
* As for the Editor's Draft at https://solidproject.org/ED/protocol , that's also being updated to better clarify (e.g., in copyright, Status of this Document) that it is not a CG report and of course continue to not imply that W3C is endorsing it. The document will live on as is, and we can cut CG-DRAFT/FINAL(s) from it.
* Lastly, I will PR with some of the other documents listed under https://solidproject.org/TR/ along these lines.

**Comment by ianbjacobs** [comment]
Hi Virginia,

I am not having luck joining the call here;
 https://www.w3.org/events/meetings/a810b828-7a85-47b8-8a6d-2359a491a7f7/20240508T140000/

Ian

> On Apr 26, 2024, at 10:21 AM, Virginia Balseiro ***@***.***> wrote:
> 
> 
> If a call would be useful, let's find time next week.
> That would be very useful, thank you! A number of us will be attending Solid Symposium next week and thus unavailable, but would you be able to join our weekly CG call on Wednesday May 8th at 14:00 UTC?
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.Message ID: ***@***.***>

--
Ian Jacobs ***@***.***>
https://www.w3.org/People/Jacobs/
Tel: +1 917 450 8783

**Comment by elf-pavlik** [comment]
You were an hour too early. It will start 20 min from now.

**Comment by elf-pavlik** [comment]
> As for the Editor's Draft at https://solidproject.org/ED/protocol , that's also being updated to better clarify (e.g., in copyright, Status of this Document) that it is not a CG report and of course continue to not imply that W3C is endorsing it. The document will live on as is, and we can cut CG-DRAFT/FINAL(s) from it.

What reason do you see for the 'ED' not to be a CG draft? I'm looking at

* https://w3c.github.io/rdf-star/cg-spec/editors_draft.html
* https://w3c.github.io/rdf-star/cg-spec

and they both seem to use the 'CG-DRAFT` template; I also notice the *Latest published version* and *Latest editor's draft* terminology. I also noticed that the 'ED' has *UNOFFICIAL* watermarks.

**Comment by ianbjacobs** [comment]
Thanks again for including me in today's meeting. Here's a quick summary of my thoughts:

* In my opinion, a good outcome is for each specification to clearly communicate both its "W3C status" and its "Solid community status." I think we can find a way to do so, with the "W3C status" being the primary set of signals, and the "Solid community status" living within that.
* For the W3C status, all specifications in the group should start as Draft Reports with the corresponding styles, copyright statement, and status information. 
* Some of these specifications may be taken up later by a Working Group. I suggest we discuss separately the value of asking contributors to make "full specification commitments" for those specifications. (The value will depend on a few factors including continuity of participation between the CG and WG.)
* For the Solid community status, during the call we started to discuss what signals the group wants to send to the community. We did not finish that discussion today, but I have a couple of observations:
   * In my view, the "Editor's Draft" label does not communicate clearly what readers/implementers should expect.
   * Please avoid terms like "Working Draft" or "Candidate Specification" that may create confusion with W3C WG signals.
   * I thought I was hearing that stability signals might be useful (like "Core" and "Experimental") but I might be mistaken.

I am happy to continue to contribute to this discussion. Thanks!

**Comment by csarven** [comment]
>What reason do you see for the 'ED' not to be a CG draft?

That might be begging the question =)

The initial intention was to not have our `ED/protocol` imply anything W3C labelled/endorsed, unless it has to do with CG-DRAFT. Right now `ED/protocol` uses the W3C ED stylesheet and the SotD is a bit mangled. It either needs to completely drop off association with W3C or as you seem to be asking about:

Our `ED/protocol` should only be unofficial CG-DRAFT. That'd be same as https://w3c.github.io/rdf-star/cg-spec/editors_draft.html AFAICT.

Our `TR/protocol` could be the CG-DRAFT once W3C approves, and then we can remove the unofficial watermark. As mentioned earlier, CG-DRAFT doesn't have to be at `TR/protocol` either. However, CG-FINAL will be published under w3.org (e.g., https://www.w3.org/2021/12/rdf-star.html )

---

Aside (speaking only based on my own understanding/observation):

As it currently stands at W3C, there is no ED or UD for CGs. While some CGs use(d) ED/UD - possibly because of process and tooling confusion, as well as nothing particularly prohibiting their use - they are incorrect and need to change. As Ian also mentioned above, CGs are only expected use CG-DRAFT/FINAL and to not communicate other kinds of document types/statuses (see also https://www.w3.org/standards/types/#x1-summary )

The tr-pages issue that I've mentioned earlier highlighted that unnecessary constraint at W3C (as I see it, IANAL) but that's not something for debate here. It is an open issue there, for W3C to consider acknowledging ED/UD for CGs because it is something used out there, and CGs can still publish CG-DRAFT/FINAL.

As I see it, if "unofficial" CG-DRAFT serves as a draft for us along the lines of what we are doing at `ED/protocol`, that'd be fine. We can treat that document as the (editor's) draft, and when we want to make it "official", we propose CG-DRAFT (without unofficial watermark) updating `TR/protocol` ("this version" referring to `https://solidproject.org/TR/2024/{YYYYMMDD}-protocol`, which is similar to https://w3c.github.io/rdf-star/cg-spec redirecting to latest CG-DRAFT at https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html ).

**Comment by elf-pavlik** [comment]
I don't think that *UNOFFICIAL* CG-DRAFT is defined somewhere. I just saw that some drafts have a *UNOFFICIAL* watermark used in their templates. Since many Solid drafts use [bikeshed](https://speced.github.io/bikeshed/#metadata) I would like to have clarity on which of the available templates should be used

![image](https://github.com/solid/specification/assets/876431/de451d23-e8bd-4118-9e32-833a8ec41a0f)

I think all drafts should use *CG-DRAFT*, which seems to add the *UNOFFICIAL DRAFT* watermark, as seen here: https://solid.github.io/data-interoperability-panel/specification/

There is also *UD* but it doesn't seem to be intended for CGs, as seen here: https://solid.github.io/data-interoperability-panel/primer/application.html

Does anyone see a problem with using the CG-DRAFT template for all CG work items, whether it is the latest published version (tagged/timestamped release snapshot) or the latest editor's draft (auto-generated from the `main` branch)? For me, this is the priority to answer because we still have time to figure out if and when to promote selected drafts to CG-FINAL.


-----

> For the Solid community status, we discussed what signals the group wants to send to the community during the call. 

As I understand the maturity levels working groups use, reaching [*CR*](https://www.w3.org/2023/Process-20231103/#RecsCR) signals that the draft is primarily stable and ready for implementation. Many people find it problematic to go that far without actual implementation experience. Solid takes a more agile approach and strives for continued implementation feedback; there is already an advanced test suite for various product classes. Having versioned releases allows better communication with developers and clarity on which draft and version are being implemented. 

For example https://communitysolidserver.github.io/CommunitySolidServer/latest/usage/notifications/
references a tagged 0.2 release of Solid Notifications Protocol.

Another example where we initially failed to do that is https://solidproject.org/TR/oidc
We had yet to tag a release before it was implemented, as we needed to make a significant change and tagged it 0.1 after that change. This made communicating much more challenging for many implementations that they are still at the pre 0.1 version.

**Comment by csarven** [comment]
>I would like to have clarity on which of the available templates should be used

W3C Team / Ian can correct me but as per https://www.w3.org/standards/types/#x2-1-w3c-community-group-report-or-w3c-business-group-report and https://www.w3.org/community/reports/reqs/ and irrespective to the tooling or environment that's used to create the report, the report needs to apply the CG-DRAFT template, styles, and include certain content (e.g., copyright, SotD).

The W3C stylesheet for CG-DRAFT by default applies the unofficial watermark, and the unofficial watermark is removed when the `no-watermark` word is a value of `<body class>` - I presume once approved by W3C Team.

https://solid.github.io/httpsig/ is already unofficial CG-DRAFT (using Respec).

I suggest all applicable Solid CG [work items](https://solidproject.org/TR/#work-items) should first transition to (unofficial) CG-DRAFT. Any work item that is deemed to be mature to be "official", we (CG) can make a request to W3C Team Process - something we currently have in the pipeline for some specs any way.

Solid CG's use of versioning is already compatible with CG Report Requirements in the Copyright line: "REPORT NAME/VERSION". That is, the reports will say CG-DRAFT/FINAL as well as can mention the specific version. This helps to distinguish different CG-DRAFTs before a CG-FINAL.

**Comment by ianbjacobs** [comment]
@elf-pavlik, the Team does not play any role in a group's decision to publish. I am joining this conversation to help with the [requirements](https://www.w3.org/community/reports/reqs/).

**Comment by elf-pavlik** [comment]
Since both ReSpec and Bikeshed add a *UNOFFICIAL* watermark, I assume this is required for any CG-DRAFT. I understand it is possible to override it and remove that watermark, but when is it appropriate to do it?

Looking again at

* https://w3c.github.io/rdf-star/cg-spec/editors_draft.html
* https://w3c.github.io/rdf-star/cg-spec

The *Latest editor's draft* does have the *UNOFFICIAL* watermark, while the *Latest published version* does not have any watermarks.

@ianbjacobs is there an official guidance for how the *UNOFFICIAL* watermark is expected to be used on a CG-DRAFT document?

**Comment by ianbjacobs** [comment]
@elf-pavlik, I believe this watermark is by design.

**Comment by elf-pavlik** [comment]
If the watermark is required, disabling it isn't a valid option. Does it mean that, for example, https://w3c.github.io/rdf-star/cg-spec doesn't meet the CG-DRAFT publishing requirements due to a lack of the *UNOFFICIAL* watermark?

**Comment by ianbjacobs** [comment]
@elf-pavlik, that appears to be a Final Report and so the expectations may be slightly different.

**Comment by csarven** [comment]
elfPavlik, AFAICT, i.e., don't quote me, both watermark and no-watermark is available from the CG-DRAFT stylesheet, so it is indeed there by design. What I'm concluding from that is that the no-watermark is applied when CG is done/ready/okay to publish the CG-DRAFT.

This is part of the clarifications that I'm seeking in https://github.com/w3c/tr-pages/issues/102 as part of the W3C documentation.

Ian, AFAICT, that is the CG-DRAFT.  CG-FINAL is at https://www.w3.org/2021/12/rdf-star.html (and that also has the `no-watermark` word in body class.)

**Comment by elf-pavlik** [comment]
https://w3c.github.io/rdf-star/cg-spec 

Ok, now I see that it says Final Report but uses the Draft Report logo
![image](https://github.com/solid/specification/assets/876431/9a1f8db7-3545-400b-a44a-2a13e51dd63e)


https://www.w3.org/2021/12/rdf-star.html

This one uses both the Final Report text and the Final Report logo.
![image](https://github.com/solid/specification/assets/876431/57f3517d-fa82-449f-af2c-1b22f2d96b0b)

**Comment by csarven** [comment]
Oh I stand corrected re https://w3c.github.io/rdf-star/cg-spec -> https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html which is indeed CG-DRAFT as far as the content is concerned. That is indeed confusing with the logo. The logo bit was probably a mistake. See also https://services.w3.org/htmldiff?doc1=https%3A%2F%2Fw3c.github.io%2Frdf-star%2Fcg-spec%2F2021-12-17.html&doc2=https%3A%2F%2Fwww.w3.org%2F2021%2F12%2Frdf-star.html which seems to be essentially identical.

**Comment by elf-pavlik** [comment]
During the call, we discussed that CG-FINAL can be updated as much as needed. Is there any disadvantage to maintaining the editor's draft as CG-DRAFT and updating CG-FINAL to the latest tagged release?

I see that *FINAL* terminology is quite confusing here, but it seems like a valid option to use it as the *Latest published*

**Comment by ianbjacobs** [comment]
@elf-pavlik, we intended "Final" to mean "he group intends for this to be referenced as a stable document (notably for IPR reasons)." If you intend to use it to signal that this publication is a stable reference, then Final seems appropriate.

**Comment by elf-pavlik** [comment]
Could you please clarify what *stable* entails?

If we draw parallels to maturity levels in a working group, can each Working Draft be considered stable, or are only the Candidate Recommendation and the later stages considered stable?

**Comment by ianbjacobs** [comment]
"Stable" here is intended to create a snapshot for IPR reasons principally. It is also intended as a signal that this particular instance is the culmination of some effort (and, for example, at this point the group would like to hand it to a Working Group). I think that if there's a constantly evolving document and the group wants to say to the community "we are still very  much working on this document" then Draft is a better fit.

**Comment by csarven** [comment]
After re-re-revisiting, comparing tool outputs, checking CG-DRAFT reports out there, reviewing style guidelines,... my current understanding is that CG-DRAFT only comes with a watermark and the no-watermark is an isolated case in https://w3c.github.io/rdf-star/cg-spec/2021-12-17.html (besides the logo being stuck on CG-DRAFT while content saying "Final"). I do not know if the use of no-watermark was allowed by W3C Team, or an error / oversight somewhere.

So, **by default**, whatever the CG wants to communicate externally should be the [versioned release](https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md#versioned-technical-report) (as per CG Guidelines) of a CG-DRAFT, e.g.:

`https://solidproject.org/TR/{YYYY}/{shortname}-{YYYYMMDD}`

and possibly the "latest published version" if only one reference is preferred:

`https://solidproject.org/TR/{shortname}`

For specs that (may likely) continue to be available from github io (GH pages) and not make their way into the solid/specification repo to be published under solidproject org's TR/ , they need to stay as CG-DRAFT's with the unofficial watermark.

---

In cases where we (CG) has an "ED" of a spec, e.g., https://solidproject.org/ED/protocol , https://solidproject.org/ED/qa , in addition to some snapshot, I think there are two routes:

Options 1:

*  switch to using [W3C's base stylesheet](https://www.w3.org/StyleSheets/TR/2021/base.css) which has no W3C markings as far as I can tell.
* update the copyright and SotD lines to disassociate itself from W3C.

Option 2:

* Stick to CG-DRAFT requirements (as it is currently in e.g., https://solidproject.org/ED/protocol ) but this URL shouldn't be linked from the W3C website or endorsed. The only exception to this is that if/when a WG wants to adopt the "latest" state of a CG specification, they can still use the ED.


From my point of view, option 2 is marginally simpler and doesn't conflict with W3C's requirements because it is published on solidproject.org - anyone can say anything about anything applies - and w3.org's CG or Report pages won't be linking to the ED any way when the actual CG-DRAFT is available / endorsed from the URLs that I mentioned above. I'm fine with option 1 too if W3C insists - it requires minor changes if/when we want to create a new CG-DRAFT out of it.

We can reflect these things in the [CG Contributing Guidelines](https://github.com/w3c-cg/solid/blob/main/CONTRIBUTING.md) once agreed.

Any way, I'm all ears if there are other / better flows.

**Comment by csarven** [comment]
@ianbjacobs , I've done the option 1 above for now (see below for details). If option 2 is also acceptable or another option is preferable, please let us know.

Can you please have a look at this CG-DRAFT report (just published following https://github.com/solid/specification/pull/651 )  and confirm that it conforms to the W3C CG-DRAFT Report requirements:

* This version: https://solidproject.org/TR/2024/protocol-20240512
* Latest published version: https://solidproject.org/TR/protocol

"This version" and "Latest published version" are borrowed from W3C terminology, so you can expect "this version" to be the immutable version (for all intents and purposes, similar to the way W3C allows revising class-1 changes to Recs) after publication and "latest published version" to be mutable (content updated whenever there is a new version).

If not, can you please let us know for any changes (either as a separate issue or in this issue would do), thanks.

I have also updated the "Editor's Draft" of Solid Protocol:

* https://solidproject.org/ED/protocol

I have it set to use the W3C Base stylesheet: https://www.w3.org/StyleSheets/TR/2021/base.css and I have updated the text (copyright, sotd, etc.) to not state that it is a W3C CG report or endorsed by W3C in any way. I want to emphasise again that this "ED" is not same as W3C ED besides using the same terms: "editor's draft".

If that is acceptable as well, we can then make sure to keep our CG-DRAFT reports and our EDs as visibly and content wise separate (not touch on "W3C") without being a bother to anyone. We (like many other CGs) have some need to publish some sort of a versioned release (e.g., CG-DRAFT, vX.Y.Z) for some of our documents as well as having an ED of sorts for internal, WIP, day-to-day stuff. Some of this discussion is also touched on https://github.com/w3c/tr-pages/issues/102

Thanks!

**Comment by ianbjacobs** [comment]
Hi @csarven, 

First, thanks for working on this. The CG-DRAFT report looks good with respect to [report requirements](https://www.w3.org/community/reports/reqs/). 

For the Editor's Draft I have some minor questions. The status section states "This report was not published by the Solid Community Group." I wasn't certain about consistency of that statement with other metadata and so would like to hear your thoughts. Specifically:

* The metadata at the top of the document says “Permission…..Assigner...W3C Solid Community Group
* Section 12.1.1 (IANA registration) says "Author/Change controller...W3C Solid Community Group"
* While I think anyone can leverage the CLA, it might be helpful to emphasize that the contributors are using this outside of any CG. Perhaps: "the Contributors to Solid Protocol, Editor’s Draft (not published by a Community Group), under the [W3C Community Contributor License Agreement (CLA)](https://www.w3.org/community/about/agreements/cla/)." Or something easier to digest.
* The Acknowledgments also talks about the community group as though the spec were published by the CG.

Thanks again for working together on this.

**Comment by elf-pavlik** [comment]
Could someone please take a look at the updated https://solid.github.io/solid-oidc/

The approach is that any version will be a CG-DRAFT as an official work item of Solid CG. There will be tagged releases with snapshots and permalinks, all of which are just versions of the same CG-DRAFT. To avoid any confusion, I won't use *Editor's draft* terminology at all. Also since this repository already uses a pull request workflow with reviews for introducing normative changes.

The linked document uses a CG-DRAFT template provided by Bikeshed. @CxRes suggested that Solid CG CG Draft shouldn't use the W3C logo, but I think this needs to be clarified if that template uses the logo by default.

**Comment by csarven** [comment]
@elf-pavlik That, irrespective to whatever tooling you're using, doesn't conform to https://www.w3.org/community/reports/reqs/ , so you'll need to update the copyright and SotD.

Removing "W3C" stuff was only in the case of CG communicating anything but CG-DRAFT/FINAL reports. So, CGs should only publish / communicate CG-DRAFT/FINAL.

**Comment by elf-pavlik** [comment]
To have SotD auto-injected by bikeshed, I may need to PR to bikeshed. This way, all SolidCG cg-drafts will have the same SotD. I will probably base it on https://github.com/speced/bikeshed/blob/main/bikeshed/spec-data/readonly/boilerplate/fedidcg/status.include

As for copyright, it indeed looks different from what I see in https://fedidcg.github.io/FedCM/

> [Copyright](https://www.w3.org/policies/#copyright) © 2024 the Contributors to the Federated Credential Management API Specification, published by the [Federated Identity Community Group](https://www.w3.org/community/fed-id/) under the [W3C Community Contributor License Agreement (CLA)](https://www.w3.org/community/about/agreements/cla/). A human-readable [summary](http://www.w3.org/community/about/agreements/cla-deed/) is available.

I'm asking on bikeshed matrix channel how we can have something similar automatically added in all Solid CG drafts.

**Comment by ianbjacobs** [comment]
@elf-pavlik, I think one can use a copyri­ght.in­clude (but I am not a bikeshed pro.)

**Comment by elf-pavlik** [comment]
Thank you, @csarven and @ianbjacobs. I have created a PR for bikeshed to include default copyright and SotD for Solid CG drafts and final drafts.

* https://github.com/speced/bikeshed/pull/2875

Once it is merged and the updated bikeshed is released, I will update all the 12 CG drafts that use bikeshed to render with that default text.

**Comment by elf-pavlik** [comment]
https://solid.github.io/solid-oidc/ has been updated to use text from https://www.w3.org/community/reports/reqs/ 

Bikeshed has the Solid CG included in https://github.com/speced/bikeshed-boilerplate/tree/main/boilerplate/solidcg

I'm in the process of updating all the remaining bikeshed-based drafts. After that I will also update the script to cut versioned releases. 

Are there any outstanding TODOs in this issue, or can we close it?

**Comment by CxRes** [comment]
I would like to express serious reservations about the changes to the copyright statement in Solid TRs that is being adopted!

I find it extremely strange how work items can be "released" under the CLA. The CLA is not a licence in the traditional sense. The CLA is an agreement between "I" the contributor and "you" The Solid Community Group (and presumably other contributor in it). In particular, "you" cannot be interpreted as citizens of world or at least Berne convention signatories. Specifically, the statement says nothing about the rights that the Solid Community Group accords to the general public to use these specifications. It follows that without a license grant, the general public has no rights to "copy" these specifications.

While I cannot speak for others, I had signed the CLA, and thus, waived my own rights over those contributions, under the assumption that my contributions will be permissively licensed by the Solid CG, i.e., my contributions will be available to the world at large, not just to a CG or its other contributors or the W3C. I expect the Solid CG to honour that expectation.
 
Therefore, I would urge the statement be redrafted. If the statement is a W3C requirement, then W3C needs to seriously revisit them, not Solid CG be asked to release work items under a more restrictive license. I have no objection to stating that contributions were made to it under the CLA and thus Solid CG has copyrights to these contributions, but it must **explicitly** state how Solid CG makes the document in question available to the world at large.

**Comment by CxRes** [comment]
> would this license work for you?

I would have, until recently. But it has a strange line in it:

> HOWEVER, the publication of derivative works of this document for use as a technical specification is expressly prohibited.

This I feel is not suitable at the CG stage, where we should be encouraging people to experiment with new ideas.

So now, I much rather prefer this: https://www.w3.org/copyright/software-license-2023/

**Comment by ianbjacobs** [comment]
Hi all, 

Our [CLA deed](https://www.w3.org/community/about/process/cla-deed/) explains what contributors give under the CLA:

<pre>
Under the CLA, each contributor gives everyone:

   * Copyright – a royalty-free license to use the copyrights for their Contributions; see section 2 of the agreement.
   * Patent – a commitment to license on a royalty-free basis their essential patent claims reading on their Contributions; see sections 3 and 12 of the agreement as well as section 9 regarding transition to the W3C Recomendation Track.
</pre>

@CxRes, you wrote "I had signed the CLA ... under the assumption that my contributions ...  will be available to the world at large". That is also my expectation per the CLA deed.

If you have any additional questions, let me know.

**Comment by CxRes** [comment]
@ianbjacobs Unfortunately, as I see it, things are not that simple...

The CLA deed that you refer to has not legal standing. It says so right in the document:
> The following is a handy, human-readable expression of key terms of the CLA. This summary is not the actual agreement. This summary has no legal value or effect and its contents do not appear in the actual agreement.

FWIW, the preceding line also makes it clear actual CLA is an agreement between "I" and "you":
> The [W3C Community Contributor License Agreement (CLA)](https://www.w3.org/community/about/agreements/cla/) expresses the full agreement between the contributors (“I” in the license agreement) and you (“you” in the license agreement).

The CLA proper does not specify the term "everyone". AFAICT, the CLA does not specify how the "you" (entity who exercises the copyright) to which "I" (the contributor)  have made the contribution, then licenses the resulting specification. In other words, there seems to be a massive gap between our shared expectations to make our contributions free and open to "everyone" and the reality of the CLA which is a limited agreement from "I" to "you", with no obligations on "you" except the attribution requirement in 2.2.

Suffice to say, this is also a deviation from the normal practice in the industry, where CLA and licenses are separate.

**Comment by ianbjacobs** [comment]
@CxRes,

The [CLA](https://www.w3.org/community/about/process/cla/) says: "12.9. You or Your. “You,” “you,” or “your” means any person or entity who exercises copyright or patent rights granted under this CLA, and any person or entity that person or entity controls."

There is no restriction to "any person" that I am aware of.

**Comment by CxRes** [comment]
@ianbjacobs 

That's not the question at all. In what way are "you", in this case the Solid CG (a member in the "any person" cohort), exercising that copyright that "I" the contributor am granting you? Are "you", for example, going to publish the specifications resulting from the "my" contributions under a permissive license or, for the sake of argument, charge someone a fee for redistributing or implementing it (Since "you" are allowed to reproduce, sublicense, distribute etc. it in any way per 2.1). The CLA holds the "I" who is signing it accountable, and not the "you" accountable in any way except 2.2. 

What I am demanding here is that the W3C Solid CG (which is a separate legal entity from any individual contributors signing the CLA, for otherwise "I" could never sign a contract i.e. the CLA, with the W3C Solid CG), the "you" in this case, in exercising the copyright that "I" have granted "you" to publishing the specification, specify to what extant it waives its rights over the said copy. Also, in general, to what standards is a W3C BG/CG supposed to hold themselves to, when they reproduce a specification developed from contributors' contributions? That is what I expect to be clearly stated in a copyright statement on every specification.

**Comment by ianbjacobs** [comment]
All participants agree upon joining the group to the CLA for their contributions to a Specification. Through the CLA, a contributor agrees to the copyright/patent/other terms for "any person or entity exercises copyright or patent rights granted under this CLA." The copyright/patent grants are available to anyone who uses the Specification. That includes the ability to make derivative works (with attribution).

How participants in the Solid community make agreements regarding contributions to work before it reaches a CG is outside of my purview.

**Comment by CxRes** [comment]
> The copyright/patent grants are available to anyone who uses the Specification.

Including the CG, which exercises that copyright when it publishes a specification. 

I am pained to stress that, just because individual contributors have waived their rights of contributions to a body of work, does not automatically imply that a particular copy of the published work is freely available (this is literally the acedemic publishers' business model). A publisher can *"without any obligation for accounting to me"*  republish/reuse the spec in any way it deems fit. 

In this circumstance and at the risk of repeating myself, I cannot help but demand that the CG (a separate entity from any individual contributor and not an "I" in the CLA) in its capacity as a publisher of these specifications explicitly specify, what claims it makes/waives on the copy it publishes in the copyright line, preferably through a license. 

Let me again stress that CG was already doing this for seven year, until about a month ago, when it abruptly switched templates. 

> How participants in the Solid community make agreements regarding contributions to work before it reaches a CG is outside of my purview.

Not at all what I am asking about! Please do not strawman the argument. This is about contributions that were made to the CG while it was publishing specifications under the MIT License.

**Comment by elf-pavlik** [comment]
@CxRes, I'm getting the impression that you are losing your patience. Please remember that everyone engaging in the discussion does their best to help.

Could you describe a concrete scenario or two that you believe would be possible under the CLA and that you would like to prevent from happening? This way, we could clarify if the suggested scenario is possible under the CLA, and if it is, is it a future or a bug?

**Comment by ianbjacobs** [comment]
@CxRes, I believe I now understand your question regarding the license of the specification as an aggregation of contributions. Let me come back to you on this.

**Comment by CxRes** [comment]
@elf-pavlik Yes and thanks for the friendly intervention. My experience coming from academia is that academics routinely waive their rights only for publishers to profit of their (typically public funded) work in increasingly devious schemes. This is not about coming up with a scenario but about the principle. If I am making a contribution to an ostensibly open source project, it is reasonable to ask that the custodians of the project will also commit to keeping the work open. By explicitly stating that it will do the right thing in black and white on top of each spec on the copyright line. Solid CG did so for 7 years and thats why I contributed. The reluctance to do so now speaks volumes.

Changing of licenses should not be done on a whim. Licenses is serious stuff with serious consequences. See the Conservancy v Vizio lawsuit, for example.

**Comment by ianbjacobs** [comment]
Hi @CxRes,

I don't think there's any reluctance. The group has been doing some cleanup, as requested by the staff. I mentioned the CLA deed earlier, which sets a very clear expectation about open licensing. If we have a glitch we will fix it with that goal in mind. Please stay tuned while I do some investigation.

**Comment by CxRes** [comment]
@ianbjacobs Sure. Thanks!

May I also point out Solid CG by its charter also requires the work items are released under W3C Software and Document License.

**Comment by ianbjacobs** [comment]
> May I also point out Solid CG by its charter also requires the work items are released under W3C Software and Document License.

I'm not sure yet whether that's relevant, but thank you for the heads-up on that.

**Comment by michielbdejong** [comment]
I think nothing stops us from dual-licensing reports. So I think the appropriate course of action would be to add a W3C license to all our work items/reports, so that we at least now start complying with our own charter, even if they already are (and possibly stay) MIT-licensed as well.

After reading Rahul’s comment I do agree with him that this statement:

> 
HOWEVER, the publication of derivative works of this document for use as a technical specification is expressly prohibited.

is a bit surprising to me, does that also hold if the technical report in question explicitly mentions its sources with attribution?

And what if this derivative work is itself W3C-licensed?

I mean, our 0.10 spec is obviously a derivative work of our 0.9 spec, right?

Can we read somewhere about why that "HOWEVER" was added and how it is to be interpreted if the derived work in question does fairly mention that it was derived from a CG report? Having people derive work from our work is, to me, a goal of our work as a CG, and not a problem to be prohibited right?

**Comment by CxRes** [comment]
@michielbdejong I can give you some context on:

> HOWEVER, the publication of derivative works of this document for use as a technical specification is expressly prohibited.

This line is from the [W3C Document License]( https://www.w3.org/copyright/document-license/) which WG's typically use. @elf-pavlik was only enquiring about my preference there.

This is not an issue of contention because the CG, by its charter, is required to use [W3C Software and Document License](https://www.w3.org/copyright/software-license-2023/) which is essentially an MIT clone to include documents and does not have any such restriction.

**Comment by ianbjacobs** [comment]
Hi @CxRes, thank you for the question and for your patience while I researched the question. I have had an opportunity to discuss this with @rigow and @dontcallmedom. One note: in what follows, my focus is on copyright licenses (not patent commitments, which are not sublicensable).

The Community Group copyright licensing system for draft Specifications involves the licensing of individual contributions under the terms set forth in the [Contributor Licensing Agreement](https://www.w3.org/community/about/process/cla/). This means a draft Specification is not licensed under one license, but under as many identical licenses as there are registered CG contributors. Once a CG specification has matured, we seek a more formal commitment on the entire Specification by inviting all participants to sign the [Final Specification Agreement](https://www.w3.org/community/about/process/final/). Those are still individual commitments, but they are scoped to the entire Specification. This is legally equivalent to a non-viral license like a BSD license or an MIT license. Because each contributor licenses individually, there is no way to have another license without exposing the licensor to unpredictable risk. Whoever puts a single license on a CG Specification would be promising rights (that they do not generally own) from other people.
